keps/sig-network/2112-netpol-namespace-name-selector/README.md

```diff
@@ -0,0 +1,371 @@
+section we outline concrete feedback in these areas.
+
+### Community feedback on the namespace.Name selector
+
+- In 2016, the argument for namespace as name selectors was first made in
+the [mailing list](https://groups.google.com/g/kubernetes-sig-network/c/GzSGt-pxBYQ/m/Rbrxve-gGgAJ), based on the fact that labels can be 
+retroactively added to namespaces pretty easily, even if your namespace isn't 
+intended to be able to send traffic somewhere: 
+
+```
+we need to clarify how namespaces are matched. I'm pretty 
+uncomfortable with using labels here, since anyone can add labels at 
+any time to any namespace and thus send traffic to your pod if they 
+know the label you're using. If it were simply a namespace name, it's 
+much easier to specifically allow only namespaces you control since you 
+know your own namespace names. 
+```
+
+- Another recent argument was made on the basis of targeting namespaces 
+using names, rather then labels, on the basis of **sheer convenience**.
+
+```
+While matching things like pods etc by label is certainly worthwhile, when 
+matching a namespace I suspect the majority of the time you only want to match 
+a single namespace. It been great match against the name rather than just a 
+label. I suspect most people don't think to add labels to the namespaces.
+```
+
+The latter comment received 9 likes as a github issue - indiciating the 
+general popularity of this as a request, and that correlates well to feedback 
+we've seen in the broader community as well.
+
+We thus conclude that, even though targeting an object using its name is not 
+normal in K8s, in the case of NetworkPolicies, the overwhelming need for 
+universal, easy to define security boundaries, makes a strong case for 
+amending the API to support a "special" selector for namespace names that is 
+**independent** of labels.
+
+### Goals
+
+- Add a `namespaceSelectorAllow` option (which is additive to the current `matchLabels` selector).
+- Allow multiple `namespaceSelectorAllow` values, so that many namespaces can be 
+allowed using the same field 
+
+### Non-Goals
+
+- Support matching of wildcard namespaces
+- Changing the internal details of how namespaceSelector works with arbitrary syntax
+- Depending on more sophisticated features (like virtual labels) to accomplish 
+this feature without changing the API (note that, if someone were 
+to write a virtualLabels KEP, however, it might change the trajectory of this 
+KEP, which is worth discussing)
+
+## Proposal
+
+In NetworkPolicy specification, inside `NetworkPolicyPeer` specify a new `namespaceSelectorAllow` field.  
+
+1) One simple way to implement this feature is to modify the NetworkPolicyPeer 
+object, like so.  Since this doesn't involve modifying a go type (i.e. were 
+not replacing a labelSelector with a different type, we expect it 
+to be a cleaner implementation:
+
+```
+type NetworkPolicyPeer struct {
+  // new field
+  namespaceSelectorAllow []string
```

Comment:
having `namespaceSelector` and `namespaceSelectorAllow` implies that the two fields differ in something having to do with "allowing", which is not accurate at all. The new field should have some name that makes it clear that it's about namespace names. (It does not need to specify "allow" because everything in `NetworkPolicyPeer` specifies something to allow.)

Reply:
yeah, namespaceSelectorAllow was a horrible choice :) , changing to `namespaceNames`

Reply:
updated